### PR TITLE
Fix assets duplicated between sibling bundles, and parallel request limit behavior

### PR DESCRIPTION
**Pull request by devongovett**
When testing in site with lots of pages, I noticed two issues:

* Assets were duplicated between sibling bundles in the same bundle group
* When hitting the parallel request limit, we were skipping creating shared bundles if any bundle group had hit the limit instead of if all bundle groups had. This meant a single bundle group could prevent other bundle groups from being optimized.

I'm having trouble reproducing these issues in a smaller case, so opening the PR for feedback.

### CODE REVIEW

File: packages/bundlers/default/src/DefaultBundler.js [hunk 1]
```diff
@@ -290,10 +306,19 @@ function deduplicateBundle(bundleGraph: MutableBundleGraph, bundle: Bundle) {
     for (let asset of assets) {
       if (
         bundle.hasAsset(asset) &&
-        bundleGraph.isAssetInAncestorBundles(bundle, asset)
+        (bundleGraph.isAssetInAncestorBundles(bundle, asset) ||
+          isAssetInSiblingBundleGroups(bundleGraph, bundle, asset))
```
Comment (wbinnssmith) [inline packages/bundlers/default/src/DefaultBundler.js]: I noticed this as well. Should we expose `isAssetInSublingBundleGroups` on the bundleGraph?

File: packages/bundlers/default/src/DefaultBundler.js [hunk 2]
```diff
@@ -258,13 +258,29 @@ export default new Bundler({
       for (let asset of assets) {
         bundleGraph.addAssetGraphToBundle(asset, sharedBundle);
         for (let bundle of sourceBundles) {
-          bundleGraph.removeAssetGraphFromBundle(asset, bundle);
+          let bundleGroups = bundleGraph.getBundleGroupsContainingBundle(
+            bundle,
+          );
+          if (
```
Comment (wbinnssmith) [inline packages/bundlers/default/src/DefaultBundler.js]: 👍 May be worth extracting this as common logic with the parallel request check above

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>packages/benchmarks/three ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 1.42m | +2.34s |
| Cached | 4.26s | +105.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/three/entry.js | 2.59mb | +0.00b | 34.00ms | +3.00ms ⚠️ |

</p></details>

<details><summary>packages/benchmarks/kitchen-sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 6.82s | -94.00ms |
| Cached | 3.61s | -49.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.d5807e82.webp | 102.94kb | +0.00b | 217.00ms | -19.00ms 🚀 |
| dist/legacy/styles.27b3555b.css | 29.00b | +0.00b | 1.47s | -79.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 4.00ms | +1.00ms ⚠️ |
| dist/legacy/index.html | 709.00b | +0.00b | 10.00ms | +1.00ms ⚠️ |
| dist/modern/kitchen-sink.89f9e289.js | 418.00b | +0.00b | 4.00ms | +1.00ms ⚠️ |

</p></details>

<details><summary>packages/benchmarks/react-hn ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 8.65s | +95.00ms |
| Cached | 3.66s | -24.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/main/index.js | 65.85kb | +0.00b | 6.00ms | -6.00ms 🚀 |
| dist/module/index.js | 37.05kb | +0.00b | 3.00ms | -4.00ms 🚀 |

</p></details>

<details><summary>packages/benchmarks/ak-editor 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/259541543369572873)**

File: packages/bundlers/default/src/DefaultBundler.js (hunk 1 above)
Comment (wbinnssmith) [reply packages/bundlers/default/src/DefaultBundler.js]: Is it also possible for siblings of ancestor bundles to include the asset? Maybe we should change `isAssetInAncestorBundles` to check siblings of ancestors as well, and change its name to reflect that.

Comment (devongovett) [reply packages/bundlers/default/src/DefaultBundler.js]: oooh maybe. I hadn't considered that... 🤔

**Pull request merged**